CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc

```diff
@@ -127,50 +130,76 @@ static function int GetSoldierInfiltration(StateObjectReference UnitRef)
 
 	UnitState = XComGameState_Unit(History.GetGameStateForObjectID(UnitRef.ObjectID));
 
-	if(!UnitState.IsSoldier())
-		return default.PERSONNEL_INFIL;
-
 	UnitInfiltration = 0;
 
-	CurrentInventory = UnitState.GetAllInventoryItems();
-	foreach CurrentInventory(InventoryItem)
-	{
-		Template = InfilMgr.GetInfilTemplateFromItem(InventoryItem.GetMyTemplateName());
-
-		if (Template == none)
-			continue;
+	// check character
+	Template = InfilMgr.GetInfilTemplateFromCharacter(UnitState.GetMyTemplate());
 
-		UnitInfiltration += (float(Template.HoursAdded) * GetUnitInfiltrationMultiplierForCategory(UnitState, InventoryItem.GetMyTemplate().ItemCat));
+	if (Template != none)
+	{
+		UnitInfiltration += Template.HoursAdded;
 	}
 
-	return int(UnitInfiltration);
-}
-
-static function float GetUnitInfiltrationMultiplierForCategory(XComGameState_Unit Unit, name category)
-{
-	local float InfiltrationMultiplier;
-	local array<XComGameState_Item>	CurrentInventory;
-	local XComGameState_Item InventoryItem;
-	local X2InfiltrationModTemplate Template;
-	local X2InfiltrationModTemplateManager InfiltrationManager;
-
-	InfiltrationManager = class'X2InfiltrationModTemplateManager'.static.GetInfilTemplateManager();
+	// loop through abilities
+	AbilityRefs = UnitState.Abilities;
+	foreach AbilityRefs(AbilityRef)
+	{
+		AbilityState = XComGameState_Ability(History.GetGameStateForObjectID(AbilityRef.ObjectID));
```

Comment:
`XComGameState_Ability` is `bTacticalTransient=true` - this fetch will always fail

Reply:
What is bTacticalTransient and how do I fix it